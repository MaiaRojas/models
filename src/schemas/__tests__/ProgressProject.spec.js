const mongoose = require('mongoose/browser');
const {
  CohortProjectSchema,
  CohortMembershipSchema,
  ProgressProjectSchema,
} = require('..')(mongoose);

describe('ProgressProjectSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, ProgressProjectSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values without ReviewAnswer', (done) => {
    const cohortProject = new mongoose.Document({}, CohortProjectSchema);
    const cohortMembership = new mongoose.Document({}, CohortMembershipSchema);

    const doc = new mongoose.Document({
      cohortProject: cohortProject._id,
      cohortMembership: cohortMembership._id,
      openedAt: new Date(),
    }, ProgressProjectSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });

  it('should successfully validate with proper values', (done) => {
    const cohortProject = new mongoose.Document({}, CohortProjectSchema);
    const cohortMembership = new mongoose.Document({}, CohortMembershipSchema);

    const doc = new mongoose.Document({
      cohortProject: cohortProject._id,
      cohortMembership: cohortMembership._id,
      openedAt: new Date(),
    }, ProgressProjectSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
