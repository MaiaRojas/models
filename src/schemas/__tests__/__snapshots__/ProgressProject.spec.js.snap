// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressProjectSchema should fail validation when missing fields are provided 1`] = `
Object {
  "cohortMembership": [ValidatorError: Path \`cohortMembership\` is required.],
  "cohortProject": [ValidatorError: Path \`cohortProject\` is required.],
}
`;
